bundle/Form/FieldTypeHandler.php

```diff
@@ -0,0 +1,112 @@
+
+    public function convertFieldValueFromForm($data)
+    {
+        return $data;
+    }
+
+    /**
+     * In most cases this will be the same as {@link self::buildUpdateFieldForm()}.
+     * For this reason default implementation falls back to the internal method
+     * {@link self::buildFieldForm()}, which should be implemented as needed.
+     */
+    public function buildFieldCreateForm(
+        FormBuilderInterface $formBuilder,
+        FieldDefinition $fieldDefinition,
+        string $languageCode
+    ): void {
+        $this->buildFieldForm($formBuilder, $fieldDefinition, $languageCode);
+    }
+
+    /**
+     * In most cases this will be the same as {@link self::buildCreateFieldForm()}.
+     * For this reason default implementation falls back to the internal method
+     * {@link self::buildFieldForm()}, which should be implemented as needed.
+     */
+    public function buildFieldUpdateForm(
+        FormBuilderInterface $formBuilder,
+        FieldDefinition $fieldDefinition,
+        Content $content,
+        string $languageCode
+    ): void {
+        $this->buildFieldForm($formBuilder, $fieldDefinition, $languageCode, $content);
+    }
+
+    /**
+     * In most cases implementations of methods {@link self::buildCreateFieldForm()}
+     * and {@link self::buildUpdateFieldForm()} will be the same, therefore default
+     * handler implementation of those falls back to this method.
+     *
+     * Implement as needed.
+     */
+    protected function buildFieldForm(
+        FormBuilderInterface $formBuilder,
+        FieldDefinition $fieldDefinition,
+        string $languageCode,
+        ?Content $content = null
+    ): void {
+        throw new RuntimeException('Not implemented.');
+    }
+
+    /**
+     * Returns default field options, created from given $fieldDefinition and $languageCode.
+     */
+    protected function getDefaultFieldOptions(
+        FieldDefinition $fieldDefinition,
+        string $languageCode,
+        ?Content $content = null
+    ): array {
+        $options = [];
+
+        $options['label'] = $fieldDefinition->getName($languageCode);
+        $options['required'] = $fieldDefinition->isRequired;
+        $options['ibexa_forms']['description'] = $fieldDefinition->getDescription($languageCode);
+        $options['ibexa_forms']['language_code'] = $languageCode;
+        $options['ibexa_forms']['fielddefinition'] = $fieldDefinition;
+
+        if ($content !== null) {
+            $options['ibexa_forms']['content'] = $content;
+        }
+
+        $options['constraints'] = [];
+        if ($fieldDefinition->isRequired) {
+            $options['constraints'][] = new Constraints\NotBlank();
+        }
+
+        return $options;
+    }
+
+    /**
+     * Adds a hidden field to the from, indicating that empty value passed
+     * for update should be ignored.
+     */
+    protected function skipEmptyUpdate(FormBuilderInterface $formBuilder, string $fieldDefinitionIdentifier): void
```

Comment:
Get rid of this here and in the consuming code (see above for the reason why).

Reply:
Fixed in 12dcfa75

Reply:
OK, but please do not fix multiple remarks in the same commit.